app/models/user.rb

```diff
@@ -9,4 +9,38 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships
+  has_many :inverse_friendships, class_name: :Friendship, foreign_key: :friend_id
+  has_many :confirmed_friendships, -> { where(confirmed: true) }, class_name: 'Friendship'
+  has_many :friends, through: :confirmed_friendships
+
+  validates :name, presence: true, length: { maximum: 40 }
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true, length: { minimum: 6 }, allow_nil: false
+  scope :all_except_current_user, ->(user) { where.not(id: user) }
+
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.confirmed }
+    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.confirmed }
+    friends_array.compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.confirmed }.compact
+  end
+
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+  end
```

Comment:
These two methods can be converted to scopes. 

Right now, you are using Ruby methods to process pending friendships and friend requests. But if you have used ActiveRecord methods, the processing would have been a lot faster as the database was created to store and query data it is always more efficient than Ruby.

- [ ] Please convert these methods to scopes.